### ISSUE TITLE
Crashing/Freeze when creating ship

### ISSUE THREAD

**Issue by KingOfNightmare** [JSJBDEV/PlatosTransporters#51]
i installed the mod both server and client side and tried to make a boat. First time it gives the book as usual but when i try to sit in the control block the game just keeps going and everything is frozen, water dosen't update anything dosen't update and then crashes. In a singleplayer world you can't exit unless you forcestop the progress. Server instead it just freezes and then after a bit it crashes the server. Will attach 2 server crash logs for more info.
[crash-2021-04-19_13.42.48-server.txt](https://github.com/JSJBDEV/PlatosTransporters/files/6336644/crash-2021-04-19_13.42.48-server.txt)
[crash-2021-04-19_13.47.27-server.txt](https://github.com/JSJBDEV/PlatosTransporters/files/6336648/crash-2021-04-19_13.47.27-server.txt)

**Comment by JSJBDEV** [comment]
Yup, I've seen this issue before but I think it's the first time I've actually been given a coherent log for it, I will have a look into it

**Comment by KingOfNightmare** [comment]
if you want more info  , in the server there are the following mod/jar

cloth-api-1.5.50
cloth-config-4.11.19-fabric
fabric-api-0.32.9+1.16.5
fabrictailor_fabric-1.5.0-1.16.5
FallingTree-Fabric-1.16.5-rc1-2.10.0
Modern-Industrializatin-0.3.16.
platos-transporters-1.16-1.7.1
RebornCore-1.6-4.7.2+build.132
RoughlyEnoughItems-5.11.202
Xaeros_Minimap_21.6.0_Fabric.1.16.5

**Comment by JSJBDEV** [comment]
When it freezes, was the ship you were working on quite big?

**Comment by KingOfNightmare** [comment]
the one in single player was like 5 oak planks, the 2 tries in server were like  15 blocks placed into a square like shape

**Comment by JSJBDEV** [comment]
yup its definitely trying to do something strange then, am looking at it now

**Comment by KingOfNightmare** [comment]
was building on dirt in  single player and i think dirt or sand in server

**Comment by JSJBDEV** [comment]
Im going to upload a new version now so you can test it, it might be fixed, its difficult to tell

**Comment by KingOfNightmare** [comment]
sure thing, will firstly test it on server side single player world then will up it on server

**Comment by JSJBDEV** [comment]
1.7.2 should be up on CF soon

**Comment by KingOfNightmare** [comment]
testing it in single player and i can make ships but  the problem is that they just go straight untill they collide with something and stop. Air ship don't even lift and just go straight, sometimes i can ear water in airships. No crash so far and no freeze .

**Comment by KingOfNightmare** [comment]
also ships move even without key in hand

**Comment by KingOfNightmare** [comment]
only wheel ships work right as intended

**Comment by JSJBDEV** [comment]
ships should now use wasd to move, all of them except wheel ships

**Comment by JSJBDEV** [comment]
they arent locked to the player because its more like the old mods, you can turn it off by pressing the freeze/unfreeze keybind

**Comment by KingOfNightmare** [comment]
pressing was just does nothing tho it only moves forward

**Comment by JSJBDEV** [comment]
do a and d turn it? they did when I tested

**Comment by JSJBDEV** [comment]
again, you should be able to turn it off with the keybind

**Comment by KingOfNightmare** [comment]
wich keybind is freeze/unfreeze?

**Comment by JSJBDEV** [comment]
ah, its called stop I think, its V by default

**Comment by JSJBDEV** [comment]
I had someone basically say some of there errors were fixed using the now normal locked mode, but if you unfreeze you can use it like you use to

**Comment by KingOfNightmare** [comment]
ok i manage to move, A and D "move" the axis but the actual model dosen't move so you have to predict were it goes
also you can't go forward and turn at the same time so you have to stand still "move" direction then go forward. A bit clucky tho (for water ships now gonna text air ships)

**Comment by JSJBDEV** [comment]
yeah ive literally just implemented for water ships so it might be a bit clunky, but you can always turn it off

**Comment by KingOfNightmare** [comment]
airships DO rotate when pressing A and D, doing it while moving stops the ship. 
Anyway this mod already is wonders! will offer you  a coffe next time i have some funds. Hope you keep supporting it

**Comment by JSJBDEV** [comment]
thank you very much! Im glad we could solve most of your problems at least, most people dont respond for days xD. I do plan on supporting Platos, (at least for fabric) until such a time that someone makes a better version with all the features I cant do

**Comment by KingOfNightmare** [comment]
wish i could help but i don't know jack shit about coding, i'm only good at 3D modeling LUL. Will keep an eye for more bugs is they show themself. p.s. how do i "elevate" with airship? as for now i'm stuck at the level i build it and if i add an anvil it will just sink into the bottom of the ocean

**Comment by JSJBDEV** [comment]
you can use I think Z by default? its in the keybind menu for ascend / descend also yeah the anvil thing is intentional xD

**Comment by PseudoDistant** [comment]
I don't know if you figured it out yet (if you have I'm sorry), in my testing it seems to be a compatibility issue with Tech Reborn. Hope that helps.
